src/utils.ts

```diff
@@ -338,13 +339,42 @@ export function isStringNotEmpty(str: string) {
     return !!str.trim();
 }
 
+export async function getElfFilePath(workspaceURI: vscode.Uri): Promise<string> {
+    let elfFilePath = "";
+    if (!workspaceURI) {
+        return elfFilePath;
+    }
+    const elfPath = path.join(workspaceURI.fsPath, "build");
+    if (!canAccessFile(elfPath, fs.constants.R_OK)) {
+        throw new Error("Build is required once to generate the ELF File");
+    }
+    const elfFiles = [];
+    fs.readdirSync(elfPath).forEach((file) => {
+        if (file.endsWith(".elf")) {
+            elfFiles.push({ label: file, description: path.join(elfPath, file) });
+        }
+    });
+    if (elfFiles.length > 1) {
+        const pickedElf = await vscode.window.showQuickPick(elfFiles, {
+            placeHolder: "Select ELF File to be use for the report generation",
```

Comment:
Instead of prompting the user, can we get the ELF file name from the project metadata file?

Reply:
Yes! that was the initial code, but changed it in case user have multiple elf file in future like build for various arch or `esp32`/ `esp32s2`
Also this would prompt user to select elf file only when there are multiple `.elf` files

Reply:
We can't have multiple ELF files for different architectures in the build directory — the build system will refuse to build for a different target without running `fullclean`. So I think we can do this automatically.